src/application/agora/widgets/proposal-handle.container.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useState } from 'react';
+
+import { useCommandQuery } from 'shared/messaging';
+
+import { getAgoraUsernameFromTwitterUsername } from '../utils';
+import { GetAgoraProposalsCommand } from '../commands';
+import { ProposalData } from '../types';
+
+import { Proposal } from './proposal';
+
+interface Properties {
+  handle: string;
+}
+
+export const ProposalHandleContainer = ({ handle }: Properties) => {
+  const [currentProposalIndex, setCurrentProposalIndex] = useState<number>(0);
+  const [currentProposal, setCurrentProposal] = useState<ProposalData>();
+
+  const agoraUsername = getAgoraUsernameFromTwitterUsername(handle);
+  const proposalQuery = useCommandQuery({
+    command: new GetAgoraProposalsCommand({
+      limit: 1,
+      offset: currentProposalIndex,
+    }),
+    enabled: agoraUsername ? agoraUsername.length > 0 : false,
+  });
+
+  const isPreviousProposalAvailable = currentProposalIndex > 0;
+  const isNextProposalAvailable = Boolean(
+    proposalQuery.data?.metadata.has_next,
+  );
+
+  const showPreviousProposal = () => {
+    if (!isPreviousProposalAvailable) {
+      return;
+    }
+
+    setCurrentProposalIndex((previous) => {
+      return previous - 1;
+    });
+  };
+
+  const showNextProposal = () => {
+    if (!isNextProposalAvailable) {
+      return;
+    }
+
+    setCurrentProposalIndex((previous) => {
+      return previous + 1;
+    });
+  };
+
+  useEffect(() => {
+    const proposalResponse = proposalQuery.data?.proposals?.at(0);
+    if (proposalResponse) {
+      setCurrentProposal(proposalResponse);
+    }
+  }, [proposalQuery.data?.proposals]);
```

Comment:
tanstack-query useQuery has extra option called `select`, by doing so you can adapt the data, I think you could do something like:
```ts
select: (response) => {{ ...response, data: { ...data, proposal: data.proposals[0] }}}
```

By doing so you can remove whole state for currentProposal and this useEffect and just use `proposalQuery.data.proposal`

Reply:
I've added the useState and useEffect to keep the value of the displayed proposal for the time of loading the next one. Is that achievable using the select param?